lib/atlas/graph_from_yaml.rb

```diff
@@ -0,0 +1,60 @@
+module Atlas
+  class GraphFromYaml
+    def self.build(graph_yaml)
+      new(graph_yaml).build_graph
+    end
+
+    def initialize(graph_yaml)
+      @graph_yaml = graph_yaml
+      @graph      = GraphBuilder.build
+    end
+
+    def build_graph
+      @graph.nodes.each do |node|
+        if node_attributes = @graph_yaml.fetch(:nodes)[node.key]
```

Comment:
Because if the `@graph_yaml` for some reason doesn't have any nodes your error message will say: `Key not found: :nodes` instead of `NoMethodError: undefined method `[]' for nil:NilClass` which is something that takes me hours to debug but that first line takes me 5 minutes.